    a. Problem Statetement

    b. Data Cleaning And Preprocessing

    c. Exploratory Data Analysis

    d. Modeling

    e. Conclusion

    f. Daftar Pustaka

Submisi Jupyter Notebook berisi kode dalam bahasa python serta 
interpretasi dan penjelasan hasil penemuan insight data dalam bentuk 
markdown.

### A. Problem Statement

<!-- - Apakah kenaikan kasus covid dapat dipicu oleh hari-hari besar?
- Apakah kasus covid untuk 2 bulan kedepannya akan mengalami kenaikan atau penurunan?
- Apakah ada pengaruh mobilitas pada kasus kenaikan covid kenaikan 


Rencana proyek
- Memberikan forecasting untuk bulan kedepannya agar masyarakat atau pemerintah bisa melakukan tindak preventif untuk mencegah kenaikan covid. -->

**Problem statement**

Pandemi covid-19 memberikan dampak yang besar terhadap laju mobilitas negara-negara, termasuk Indonesia. Hal itu disebabkan dengan kenaikan kasus covid yang terus menerus serta varian covid baru yang bermunculan. Mobilitas yang terhampat dapat mempengaruhi berbagai aspek, seperti : ekonomi, sosial, kesehatan, dan lain-lain. Oleh karena itu perlu adanya tindakan yang tepat untuk menanggulangi kasus covid 19.


**Tujuan**
* Memberikan hasil Analisis untuk mencari pola kenaikan kasus covid
* Menentukan apakah kasus covid dapat dipicu oleh hari-hari besar 
* Melakukan prediksi untuk beberapa bulan selanjutnya agar pemerintah atau masyarakat dapat memberikan langkah preventif

**Rencana proyek**
* Melakukan EDA
- Melakukan prediksi beberapa bulan kedepan menggunakan algoritma sarimax

### B. Data Cleaning And Preprocessing

In [206]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import numpy as np
plt.style.use('fivethirtyeight')

#### **a. Dataset Daily Update Data Agregat Covid-19 Jakarta**
pada dataset daily update terdapat 13 sheet, tetapi kami hanya menggunakan 5 sheet saja

##### Data Jakarta

In [5]:
df_DU_Jakarta = pd.read_excel('Daily Update Data Agregat Covid-19 Jakarta.xlsx', sheet_name='Data Jakarta')
df_DU_Jakarta.shape

(838, 14)

In [6]:
# check dataframe for missing values
df_DU_Jakarta.isnull().sum()

Tanggal              0
Jam                838
Tanggal Jam        838
Meninggal            0
Sembuh               0
Self Isolation       0
Masih Perawatan      0
Total Pasien         0
Positif Harian       0
Positif Aktif        0
Sembuh Harian        0
Tanpa Gejala       140
Bergejala          140
Belum Ada Data     140
dtype: int64

pada variabel jam dan tanggal jam memiliki keseluruhan data null sehingga dilakukan *dropping*, kemudian pada variabel tanpa gejala, bergejala, dan belum ada data tidak dilakukan *dropping* karena kami asumsikan akan menghilangkan informasi penting sehingga kami lakukan imputasi dengan mengisinya dengan nilai 0

In [7]:
df_DU_Jakarta.drop(['Jam','Tanggal Jam'],axis=1,inplace=True)
df_DU_Jakarta.fillna(0,inplace=True)

In [8]:
# check dataframe for information about data types
df_DU_Jakarta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 838 entries, 0 to 837
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Tanggal          838 non-null    datetime64[ns]
 1   Meninggal        838 non-null    int64         
 2   Sembuh           838 non-null    int64         
 3   Self Isolation   838 non-null    int64         
 4   Masih Perawatan  838 non-null    int64         
 5   Total Pasien     838 non-null    int64         
 6   Positif Harian   838 non-null    int64         
 7   Positif Aktif    838 non-null    int64         
 8   Sembuh Harian    838 non-null    int64         
 9   Tanpa Gejala     838 non-null    float64       
 10  Bergejala        838 non-null    float64       
 11  Belum Ada Data   838 non-null    float64       
dtypes: datetime64[ns](1), float64(3), int64(8)
memory usage: 78.7 KB


Karena Tanggal sudah bertipe datetime sehingga tidak perlu dilakukan preprocessing

In [9]:
df_DU_Jakarta

Tanggal  Meninggal   Sembuh  Self Isolation  Masih Perawatan  \
0   2020-03-01          0        0               0                0   
1   2020-03-02          0        0               0                0   
2   2020-03-03          1        0               0                2   
3   2020-03-04          1        0               0                2   
4   2020-03-05          3        0               0                4   
..         ...        ...      ...             ...              ...   
833 2022-06-12      15303  1236002            2000              214   
834 2022-06-13      15304  1236184            2179              200   
835 2022-06-14      15304  1236374            2446              260   
836 2022-06-15      15304  1236528            2980              302   
837 2022-06-16      15305  1236820            3346              339   

     Total Pasien  Positif Harian  Positif Aktif  Sembuh Harian  Tanpa Gejala  \
0               0               0              0              0           0.0   
1               0               0              0              0           0.0   
2               3               3              2              0           0.0   
3               3               0              2              0           0.0   
4               7               4              4              0           0.0   
..            ...             ...            ...            ...           ...   
833       1253519             322           2214            164           0.0   
834       1253867             348           2379            182           0.0   
835       1254384             517           2706            190           0.0   
836       1255114             730           3282            154           0.0   
837       1255810             696           3685            292           0.0   

     Bergejala  Belum Ada Data  
0          0.0             0.0  
1          0.0             0.0  
2          0.0             0.0  
3          0.0             0.0  
4          0.0             0.0  
..         ...             ...  
833        9.0          2205.0  
834       14.0          2365.0  
835       14.0          2692.0  
836       15.0          3267.0  
837       14.0          3671.0  

[838 rows x 12 columns]

##### Daily Vaksinasi Wilayah

In [15]:
#vaksinasi wilayah
df_DU_DVW = pd.read_excel('Daily Update Data Agregat Covid-19 Jakarta.xlsx', sheet_name='Daily Vaksinasi Wilayah').iloc[:,:3].dropna(axis=0, how='any')
df_DU_DVW

Tanggal  Cakupan Suntik 1  Cakupan Suntik 2
0   2021-06-12           19281.0            1173.0
1   2021-06-13             198.0              29.0
2   2021-06-14           79491.0            3691.0
3   2021-06-15           81545.0            3385.0
4   2021-06-16           92210.0            3436.0
..         ...               ...               ...
364 2022-06-11             430.0             280.0
365 2022-06-12             300.0             194.0
366 2022-06-13             900.0            1272.0
367 2022-06-14            1028.0            1112.0
368 2022-06-15            1141.0            1003.0

[369 rows x 3 columns]

In [14]:
df_DU_DVW.isna().sum()

Tanggal             0
Cakupan Suntik 1    0
Cakupan Suntik 2    0
dtype: int64

In [13]:
df_DU_DVW.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 369 entries, 0 to 368
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Tanggal           369 non-null    datetime64[ns]
 1   Cakupan Suntik 1  369 non-null    float64       
 2   Cakupan Suntik 2  369 non-null    float64       
dtypes: datetime64[ns](1), float64(2)
memory usage: 11.5 KB


##### Vaksinasi Wilayah

In [29]:
#vaksinasi wilayah
df_DU_VW = pd.read_excel('Daily Update Data Agregat Covid-19 Jakarta.xlsx', sheet_name='Vaksinasi Wilayah',skiprows=1)
df_DU_VW.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44 entries, 0 to 43
Columns: 1118 entries, No. to Cakupan Suntik 2.370
dtypes: float64(4), int64(742), object(372)
memory usage: 384.4+ KB


In [44]:
df_DU_VW.isna().sum().sum()

0

In [30]:
df_DU_VW.head()

No.    Kecamatan Kota / Kabupaten  Target Total per Kecamatan  \
0    1       Cakung    Jakarta Timur                      515628   
1    2   Cengkareng    Jakarta Barat                      519118   
2    3   Kali Deres    Jakarta Barat                      406018   
3    4    Cilincing    Jakarta Utara                      392136   
4    5  Duren Sawit    Jakarta Timur                      404329   

   Target Harian per Kecamatan  Cakupan Suntik 1      Keterangan  \
0                         5333             74966  Tidak Tercapai   
1                         5041             85668  Tidak Tercapai   
2                         4295             42991  Tidak Tercapai   
3                         4073             41494  Tidak Tercapai   
4                         4006             41413  Tidak Tercapai   

   Cakupan Suntik 2  Cakupan Suntik 1.1    Keterangan.1  ...  \
0             37590                 897  Tidak Tercapai  ...   
1             47581                 429  Tidak Tercapai  ...   
2             23222                 162  Tidak Tercapai  ...   
3             22305                  17  Tidak Tercapai  ...   
4             28993                   7  Tidak Tercapai  ...   

   Cakupan Suntik 2.367  Cakupan Suntik 1.368  Keterangan.368  \
0                    41                    35  Tidak Tercapai   
1                    48                    29  Tidak Tercapai   
2                    25                    44  Tidak Tercapai   
3                    76                    38  Tidak Tercapai   
4                    31                    26  Tidak Tercapai   

   Cakupan Suntik 2.368  Cakupan Suntik 1.369  Keterangan.369  \
0                    46                    56  Tidak Tercapai   
1                    54                    33  Tidak Tercapai   
2                    29                    25  Tidak Tercapai   
3                    57                    64  Tidak Tercapai   
4                    27                    32  Tidak Tercapai   

   Cakupan Suntik 2.369  Cakupan Suntik 1.370 Keterangan.370  \
0                    48                   NaN            NaN   
1                    35                   NaN            NaN   
2                    28                   NaN            NaN   
3                    52                   NaN            NaN   
4                    25                   NaN            NaN   

   Cakupan Suntik 2.370  
0                   NaN  
1                   NaN  
2                   NaN  
3                   NaN  
4                   NaN  

[5 rows x 1118 columns]

In [31]:
df_DU_VW = df_DU_VW.drop(df_DU_VW.select_dtypes('float64'),axis=1)

In [32]:
df_DU_VW_grouped_kec = df_DU_VW.groupby(["No.","Kecamatan","Kota / Kabupaten","Target Total per Kecamatan","Target Harian per Kecamatan"]).sum().sum(axis=1).to_frame().rename(columns={0:'Total Vaksinasi'}).reset_index().drop(['No.','Target Total per Kecamatan','Target Harian per Kecamatan'],axis=1)
df_DU_VW_grouped_kec

Kecamatan  Kota / Kabupaten  Total Vaksinasi
0                Cakung     Jakarta Timur           555303
1            Cengkareng     Jakarta Barat           694473
2            Kali Deres     Jakarta Barat           587065
3             Cilincing     Jakarta Utara           483607
4           Duren Sawit     Jakarta Timur           470542
5         Tanjung Priok     Jakarta Utara           782796
6             Jagakarsa   Jakarta Selatan           406610
7           Kebon Jeruk     Jakarta Barat           485189
8                  Koja     Jakarta Utara           608483
9        Kebayoran Lama   Jakarta Selatan           461942
10           Jatinegara     Jakarta Timur           517532
11          Kramat Jati     Jakarta Timur           461583
12         Pasar Minggu   Jakarta Selatan           421652
13          Penjaringan     Jakarta Utara           631374
14              Ciracas     Jakarta Timur           354163
15          Pulo Gadung     Jakarta Timur           408424
16            Kembangan     Jakarta Barat           468783
17             Cipayung     Jakarta Timur           652380
18              Tambora     Jakarta Barat           363288
19         Pesanggrahan   Jakarta Selatan           467896
20            Kemayoran     Jakarta Pusat           633835
21                Tebet   Jakarta Selatan           336648
22    Grogol Petamburan     Jakarta Barat           532183
23             Palmerah     Jakarta Barat           530845
24           Pasar Rebo     Jakarta Timur           425287
25             Cilandak   Jakarta Selatan           510094
26              Makasar     Jakarta Timur           428191
27             Matraman     Jakarta Timur           448440
28          Tanah Abang     Jakarta Pusat          1057709
29             Pancoran   Jakarta Selatan           382371
30           Pademangan     Jakarta Utara           384070
31     Mampang Prapatan   Jakarta Selatan           271122
32       Kebayoran Baru   Jakarta Selatan          1898024
33           Taman Sari     Jakarta Barat           227378
34          Sawah Besar     Jakarta Pusat           441525
35        Kelapa Gading     Jakarta Utara           458017
36           Johar Baru     Jakarta Pusat           351280
37                Senen     Jakarta Pusat           364070
38           Setia Budi   Jakarta Selatan           407826
39        Cempaka Putih     Jakarta Pusat           225989
40               Gambir     Jakarta Pusat           525019
41              Menteng     Jakarta Pusat           306413
42    Kep. Seribu Utara  Kepulauan Seribu            24283
43  Kep. Seribu Selatan  Kepulauan Seribu            21636

In [33]:
df_DU_VW_grouped_kota = df_DU_VW_grouped_kec.copy()
df_DU_VW_grouped_kota = df_DU_VW_grouped_kota.reset_index().drop(['Kecamatan','index'], axis=1).groupby(['Kota / Kabupaten']).sum()
df_DU_VW_grouped_kota

Total Vaksinasi
Kota / Kabupaten                 
Jakarta Barat             3889204
Jakarta Pusat             3905840
Jakarta Selatan           5564185
Jakarta Timur             4721845
Jakarta Utara             3348347
Kepulauan Seribu            45919

##### Vaksinasi Booster Wilayah

In [35]:
df_DU_VBW = pd.read_excel('Daily Update Data Agregat Covid-19 Jakarta.xlsx', sheet_name='Vaksinasi Booster Wilayah')
df_DU_VBW

Tanggal      Kecamatan          Wilayah  Akumulasi Cakupan D3  \
0    2022-02-06         Cakung    Jakarta Timur                 12745   
1    2022-02-06     Cengkareng    Jakarta Barat                 16789   
2    2022-02-06     Kali Deres    Jakarta Barat                 12736   
3    2022-02-06      Cilincing    Jakarta Utara                  5756   
4    2022-02-06    Duren Sawit    Jakarta Timur                 20377   
...         ...            ...              ...                   ...   
5715 2022-06-15     Taman Sari    Jakarta Barat                 61254   
5716 2022-06-15        Tambora    Jakarta Barat                 57592   
5717 2022-06-15    Tanah Abang    Jakarta Pusat                 71266   
5718 2022-06-15  Tanjung Priok    Jakarta Utara                130737   
5719 2022-06-15          Tebet  Jakarta Selatan                 60507   

      Persentase D3 terhadap D2 (Masyarakat Rentan/Umum)  \
0                                                0.0672    
1                                                0.0601    
2                                                0.0516    
3                                                0.0272    
4                                                0.1087    
...                                                 ...    
5715                                             0.5997    
5716                                             0.3853    
5717                                             0.2584    
5718                                             0.3808    
5719                                             0.4225    

      Persentase D3 terhadap D2  
0                        0.0460  
1                        0.0513  
2                        0.0442  
3                        0.0243  
4                        0.0853  
...                         ...  
5715                     0.5287  
5716                     0.3315  
5717                     0.2035  
5718                     0.3286  
5719                     0.3396  

[5720 rows x 6 columns]

In [40]:
df_DU_VBW.isna().sum()

Tanggal                                               0
Kecamatan                                             0
Wilayah                                               0
Akumulasi Cakupan D3                                  0
Persentase D3 terhadap D2 (Masyarakat Rentan/Umum)    0
Persentase D3 terhadap D2                             0
dtype: int64

In [41]:
df_DU_VBW.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5720 entries, 0 to 5719
Data columns (total 6 columns):
 #   Column                                              Non-Null Count  Dtype         
---  ------                                              --------------  -----         
 0   Tanggal                                             5720 non-null   datetime64[ns]
 1   Kecamatan                                           5720 non-null   object        
 2   Wilayah                                             5720 non-null   object        
 3   Akumulasi Cakupan D3                                5720 non-null   int64         
 4   Persentase D3 terhadap D2 (Masyarakat Rentan/Umum)  5720 non-null   float64       
 5   Persentase D3 terhadap D2                           5720 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(2)
memory usage: 268.2+ KB


In [39]:
df_DU_VBW_grouped = df_DU_VBW.groupby(['Tanggal','Wilayah']).agg({'Akumulasi Cakupan D3':'sum','Persentase D3 terhadap D2 (Masyarakat Rentan/Umum)':'mean','Persentase D3 terhadap D2':'mean'}).reset_index()
df_DU_VBW_grouped

Tanggal           Wilayah  Akumulasi Cakupan D3  \
0   2022-02-06     Jakarta Barat                124888   
1   2022-02-06     Jakarta Pusat                184916   
2   2022-02-06   Jakarta Selatan                227642   
3   2022-02-06     Jakarta Timur                184524   
4   2022-02-06     Jakarta Utara                 91900   
..         ...               ...                   ...   
775 2022-06-15     Jakarta Pusat                689767   
776 2022-06-15   Jakarta Selatan                976313   
777 2022-06-15     Jakarta Timur                944297   
778 2022-06-15     Jakarta Utara                579402   
779 2022-06-15  Kepulauan Seribu                 15908   

     Persentase D3 terhadap D2 (Masyarakat Rentan/Umum)  \
0                                             0.086900    
1                                             0.151487    
2                                             0.114440    
3                                             0.105720    
4                                             0.072667    
..                                                 ...    
775                                           0.469450    
776                                           0.389460    
777                                           0.465350    
778                                           0.410350    
779                                           0.887350    

     Persentase D3 terhadap D2  
0                     0.068612  
1                     0.107938  
2                     0.086190  
3                     0.083080  
4                     0.055100  
..                         ...  
775                   0.398838  
776                   0.329740  
777                   0.392510  
778                   0.338600  
779                   0.681050  

[780 rows x 5 columns]

##### Data Input Daily Vaksinasi Anak

In [47]:
df_DU_Vaks_Anak = pd.read_excel('Daily Update Data Agregat Covid-19 Jakarta.xlsx', sheet_name='Data Input Daily Vaksinasi Anak')
df_DU_Vaks_Anak

Tanggal  Dosis  JAKARTA PUSAT  JAKARTA UTARA  JAKARTA BARAT  \
0   2021-12-14      1           1860           1447           1738   
1   2021-12-15      1           4366           6742           6466   
2   2021-12-16      1           6981           9442           5807   
3   2021-12-17      1           6171           9621           7251   
4   2021-12-18      1            821            400            106   
..         ...    ...            ...            ...            ...   
345 2022-06-13      2             11             34             17   
346 2022-06-14      1             13             29             21   
347 2022-06-14      2             20             53             36   
348 2022-06-15      1             13             21             19   
349 2022-06-15      2             27             33             23   

     JAKARTA SELATAN  JAKARTA TIMUR  ADM. KEPULAUAN SERIBU  Total  
0               2099           2440                    105   9689  
1               6810           7022                    451  31857  
2               6638           9279                    595  38742  
3               6009          11995                    393  41440  
4                536            760                     10   2633  
..               ...            ...                    ...    ...  
345               27             34                      0    123  
346               21             32                      0    116  
347               42             54                      0    205  
348               29             43                      0    125  
349               57             72                      0    212  

[350 rows x 9 columns]

In [48]:
df_DU_Vaks_Anak.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 350 entries, 0 to 349
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Tanggal                350 non-null    datetime64[ns]
 1   Dosis                  350 non-null    int64         
 2   JAKARTA PUSAT          350 non-null    int64         
 3   JAKARTA UTARA          350 non-null    int64         
 4   JAKARTA BARAT          350 non-null    int64         
 5   JAKARTA SELATAN        350 non-null    int64         
 6   JAKARTA TIMUR          350 non-null    int64         
 7   ADM. KEPULAUAN SERIBU  350 non-null    int64         
 8   Total                  350 non-null    int64         
dtypes: datetime64[ns](1), int64(8)
memory usage: 24.7 KB


In [49]:
df_DU_Vaks_Anak.isna().sum()

Tanggal                  0
Dosis                    0
JAKARTA PUSAT            0
JAKARTA UTARA            0
JAKARTA BARAT            0
JAKARTA SELATAN          0
JAKARTA TIMUR            0
ADM. KEPULAUAN SERIBU    0
Total                    0
dtype: int64

#### **b. Dataset Mobilitas**

In [62]:
df_M = pd.read_csv('2020_ID_Region_Mobility_Report.csv')
df_M.shape

(11235, 15)

In [63]:
df_M.isna().sum()

country_region_code                                       0
country_region                                            0
sub_region_1                                            321
sub_region_2                                          11235
metro_area                                            11235
iso_3166_2_code                                         321
census_fips_code                                      11235
place_id                                                  0
date                                                      0
retail_and_recreation_percent_change_from_baseline       50
grocery_and_pharmacy_percent_change_from_baseline        99
parks_percent_change_from_baseline                      118
transit_stations_percent_change_from_baseline           391
workplaces_percent_change_from_baseline                   0
residential_percent_change_from_baseline                  0
dtype: int64

In [64]:
df_M.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11235 entries, 0 to 11234
Data columns (total 15 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   country_region_code                                 11235 non-null  object 
 1   country_region                                      11235 non-null  object 
 2   sub_region_1                                        10914 non-null  object 
 3   sub_region_2                                        0 non-null      float64
 4   metro_area                                          0 non-null      float64
 5   iso_3166_2_code                                     10914 non-null  object 
 6   census_fips_code                                    0 non-null      float64
 7   place_id                                            11235 non-null  object 
 8   date                                                11235 non-null  object 


In [65]:
df_M.drop(['sub_region_2','metro_area','census_fips_code','place_id'],axis=1,inplace=True)

In [66]:
df_M

country_region_code country_region  sub_region_1 iso_3166_2_code  \
0                      ID      Indonesia           NaN             NaN   
1                      ID      Indonesia           NaN             NaN   
2                      ID      Indonesia           NaN             NaN   
3                      ID      Indonesia           NaN             NaN   
4                      ID      Indonesia           NaN             NaN   
...                   ...            ...           ...             ...   
11230                  ID      Indonesia  West Sumatra           ID-SB   
11231                  ID      Indonesia  West Sumatra           ID-SB   
11232                  ID      Indonesia  West Sumatra           ID-SB   
11233                  ID      Indonesia  West Sumatra           ID-SB   
11234                  ID      Indonesia  West Sumatra           ID-SB   

             date  retail_and_recreation_percent_change_from_baseline  \
0      2020-02-15                                               -2.0    
1      2020-02-16                                               -3.0    
2      2020-02-17                                               -3.0    
3      2020-02-18                                               -3.0    
4      2020-02-19                                               -3.0    
...           ...                                                ...    
11230  2020-12-27                                                4.0    
11231  2020-12-28                                                2.0    
11232  2020-12-29                                                1.0    
11233  2020-12-30                                               11.0    
11234  2020-12-31                                               17.0    

       grocery_and_pharmacy_percent_change_from_baseline  \
0                                                   -2.0   
1                                                   -3.0   
2                                                   -4.0   
3                                                   -2.0   
4                                                   -4.0   
...                                                  ...   
11230                                               28.0   
11231                                               25.0   
11232                                               26.0   
11233                                               38.0   
11234                                               50.0   

       parks_percent_change_from_baseline  \
0                                    -8.0   
1                                    -7.0   
2                                    -7.0   
3                                    -4.0   
4                                    -3.0   
...                                   ...   
11230                                29.0   
11231                                32.0   
11232                                32.0   
11233                                53.0   
11234                                51.0   

       transit_stations_percent_change_from_baseline  \
0                                                1.0   
1                                               -4.0   
2                                               -1.0   
3                                                1.0   
4                                                0.0   
...                                              ...   
11230                                           -8.0   
11231                                           -3.0   
11232                                            2.0   
11233                                            5.0   
11234                                           10.0   

       workplaces_percent_change_from_baseline  \
0                                            5   
1                                            2   
2                                            3   
3                                            2   
4                                            1   
...  

### C. Exploratory Data Analysis

#### **a. Dataset Daily Update Data Agregat Covid-19 Jakarta**

##### Data Jakarta

In [68]:
df_DU_Jakarta

Tanggal  Meninggal   Sembuh  Self Isolation  Masih Perawatan  \
0   2020-03-01          0        0               0                0   
1   2020-03-02          0        0               0                0   
2   2020-03-03          1        0               0                2   
3   2020-03-04          1        0               0                2   
4   2020-03-05          3        0               0                4   
..         ...        ...      ...             ...              ...   
833 2022-06-12      15303  1236002            2000              214   
834 2022-06-13      15304  1236184            2179              200   
835 2022-06-14      15304  1236374            2446              260   
836 2022-06-15      15304  1236528            2980              302   
837 2022-06-16      15305  1236820            3346              339   

     Total Pasien  Positif Harian  Positif Aktif  Sembuh Harian  Tanpa Gejala  \
0               0               0              0              0           0.0   
1               0               0              0              0           0.0   
2               3               3              2              0           0.0   
3               3               0              2              0           0.0   
4               7               4              4              0           0.0   
..            ...             ...            ...            ...           ...   
833       1253519             322           2214            164           0.0   
834       1253867             348           2379            182           0.0   
835       1254384             517           2706            190           0.0   
836       1255114             730           3282            154           0.0   
837       1255810             696           3685            292           0.0   

     Bergejala  Belum Ada Data  
0          0.0             0.0  
1          0.0             0.0  
2          0.0             0.0  
3          0.0             0.0  
4          0.0             0.0  
..         ...             ...  
833        9.0          2205.0  
834       14.0          2365.0  
835       14.0          2692.0  
836       15.0          3267.0  
837       14.0          3671.0  

[838 rows x 12 columns]

In [84]:
fig = px.line(x='Tanggal', y=['Total Pasien','Meninggal','Sembuh'], data_frame=df_DU_Jakarta,title='Total Pasien Covid, Pasien Meninggal, dan Pasien Sembuh pada Maret 2020 hinga Juni 2022')
fig.add_annotation(x='2021-06-20', y=680000,
            text="Terjadi Lonjakan covid",
            align="center",
            arrowhead=2,
            arrowsize=1,
            arrowwidth=2,
            # yshift=40,
            # xshift=10,
            ax=-50,
            ay=-30,
            bordercolor="#333",
            borderwidth=2,
            borderpad=4,
            bgcolor="#F15412",
            opacity=0.8,
            font_color="#fff")

fig.add_annotation(x='2022-02-01', y=1100000,
            text="Terjadi Lonjakan covid",
            align="center",
            arrowhead=2,
            arrowsize=1,
            arrowwidth=2,
            # yshift=40,
            # xshift=10,
            ax=-50,
            ay=-30,
            bordercolor="#333",
            borderwidth=2,
            borderpad=4,
            bgcolor="#F15412",
            opacity=0.8,
            font_color="#fff")
fig.show()

Interpretasi

Pada diagram garis diatas menampilkan berkembangan jumlah pasien yang terkena covid, pasien yang meninggal, dan pasien yang sembuh dari tahun 2020 bulan maret hingga tahun 2022 bulan Juni.
Lalu dapat dilihat bahwa terjadi 2 kenaikan yang signifikan. Kenaikan pertama pada tahun 2021 bulan Juni akhir hingga bulan juli akhir  dan kenaikan kedua pada awal febuari 2022 hingga akhir febuari.

Asumsi kami pada kenaikan pertama hal tersebut dikarenakan pada pada tanggal 20 Juli 2021 terdapat idul adha sehingga terjadi arus mudik dan arus balik. Lalu terdapat varian baru seperti Alpha, Beta, dan Delta. Hal ini juga didukung oleh berita [suara.com](https://www.suara.com/news/2021/06/27/003810/covid-19-di-indonesia-menggila-ini-data-lonjakan-kasus-sepanjang-bulan-juni). Kemudian untuk kenaikan kedua pada awal febuari terjadi karena gelombang kedua covid untuk varian omicron berdasarkan berita [nasional.kontan.co.id](https://nasional.kontan.co.id/news/kasus-covid-19-24-februari-2022-tambah-57426-tapi-ada-kabar-baik)

In [201]:
px.line(x='Tanggal', y=['Positif Harian'], data_frame=df_DU_Jakarta,title='Data Pasien covid Positif Harian',color_discrete_sequence=['red'])

Interpretasi

Hal tersebut dapat dilihat lebih jelas pada data covid positif harian yang memiliki kenaikan pada bulan juli dan febuari.

In [100]:
print('Data positif covid tanpa gejala         : ',df_DU_Jakarta['Tanpa Gejala'].sum())
print('Data positif covid dengan gejala        : ',df_DU_Jakarta['Bergejala'].sum())
print('Data positif covid yang tidak diketahui : ',df_DU_Jakarta['Belum Ada Data'].sum())

Data positif covid tanpa gejala         :  2584819.0
Data positif covid dengan gejala        :  3501548.0
Data positif covid yang tidak diketahui :  5389258.0


In [127]:
df_DU_Jakarta[['Tanpa Gejala','Bergejala','Belum Ada Data']].sum().sum()

11475625.0

In [184]:
fig = px.pie(names=['Tanpa Gejala','Bergejala','Belum Ada Data'],values=df_DU_Jakarta[['Tanpa Gejala','Bergejala','Belum Ada Data']].sum().values,title='Data Pasien covid Positif Harian', color_discrete_sequence=px.colors.sequential.RdBu,hole=.4)
fig.update_traces(textposition='inside', textinfo='percent+label', marker=dict(line=dict(color='#000000', width=2)))
fig.update_layout(
    # Add annotations in the center of the donut pies.
    annotations=[dict(text='Total :', x=0.5, y=0.56, font_size=15, showarrow=False),
                 dict(text='11475625.0', x=0.5, y=0.46, font_size=13, showarrow=False)])
fig.show()

Interpretasi

Dari 11475625 pasien covid 30.5% mengalami gejala dan 22.5% pasien covid tidak mengalami gejala.

dari perbedaan yang sedikit tersebut yaitu 8%, kami asumsikan gejala tidak dapat menjadi faktor utama untuk mengetahui bahwa apakah seorang itu mengalami covid atau tidak.

In [187]:
df_DU_Jakarta[['Sembuh','Meninggal']].sum()

Sembuh       407740901
Meninggal      6277939
dtype: int64

In [194]:
df_DU_Jakarta[['Sembuh','Meninggal']].sum().sum()

414018840

In [200]:
fig = px.pie(names=['Sembuh','Meninggal'],values=[407740901,6277930],title='Persentase Kesembuhan Covid',color_discrete_sequence=px.colors.sequential.RdBu,hole=.4)
fig.update_traces(textinfo='percent+label', marker=dict(line=dict(color='#000000', width=2)))
fig.update_layout(
    # Add annotations in the center of the donut pies.
    annotations=[dict(text='Total :', x=0.5, y=0.56, font_size=15, showarrow=False),
                 dict(text='414018840', x=0.5, y=0.46, font_size=13, showarrow=False)])
fig.show()

Interpretasi

Dari 414018840 Pasien 98.48% pasien covid sembuh dan 1.52% pasien meninggal dengan angka kesembuhan pasien yang tinggi kami asumsikan bahwa covid bukanlah penyakit yang mematikan, tetapi memiliki penularan yang tinggi oleh karena itu mesti tetap waspada

##### Daily Vaksinasi Wilayah

In [202]:
df_DU_DVW

Tanggal  Cakupan Suntik 1  Cakupan Suntik 2
0   2021-06-12           19281.0            1173.0
1   2021-06-13             198.0              29.0
2   2021-06-14           79491.0            3691.0
3   2021-06-15           81545.0            3385.0
4   2021-06-16           92210.0            3436.0
..         ...               ...               ...
364 2022-06-11             430.0             280.0
365 2022-06-12             300.0             194.0
366 2022-06-13             900.0            1272.0
367 2022-06-14            1028.0            1112.0
368 2022-06-15            1141.0            1003.0

[369 rows x 3 columns]

In [203]:
px.line(x='Tanggal',y=['Cakupan Suntik 1','Cakupan Suntik 2'],data_frame=df_DU_DVW,title='Cakupan Suntik Covid-19 Jakarta')

In [204]:
df_DU_DVW[['Cakupan Suntik 1','Cakupan Suntik 2']].sum()

Cakupan Suntik 1    8594205.000
Cakupan Suntik 2    8207295.023
dtype: float64

In [240]:
df_DU_DVW[['Cakupan Suntik 1','Cakupan Suntik 2']].sum().sum()

16801500.023000002

In [255]:
# import plotly.graph_objs as go

# data = [
#     # Portfolio (inner donut)
#     go.Pie(values=[8594205,8207295,261950861],
#     labels=['Reds','Blues','Greens'],
#     domain={'x':[0.2,0.8], 'y':[0.1,0.9]},
#     hole=0.5,
#     direction='clockwise',
#     sort=False,
#     marker={'colors':['#CB4335','#2E86C1']}),
    
#     # Individual components (outer donut)
#     go.Pie(values=[16801500,278752361],
#     labels=['Medium Red','Light Red','Medium Blue','Light Blue'],
#     domain={'x':[0.1,0.9], 'y':[0,1]},
#     hole=0.75,
#     direction='clockwise',
#     sort=False,
#     marker={'colors':['#EC7063','#85C1E9','#F1948A','#5DADE2']},
#     showlegend=False)
# ]

# fig = go.Figure(data=data, layout={'title':'Nested Pie Chart'})
# fig.show()

In [260]:
fig = px.pie(names=['Cakupan Suntik 1 ','Cakupan Suntik 2 ','Total Penduduk Indonesia'],values=[8594205,8207295,278752361],title='Persentase Kesembuhan Covid',color_discrete_sequence=px.colors.sequential.RdBu,hole=.4)
fig.update_traces(textinfo='percent+label', marker=dict(line=dict(color='#000000', width=2)))
fig.update_layout(
    # Add annotations in the center of the donut pies.
    annotations=[dict(text='Total :', x=0.5, y=0.56, font_size=15, showarrow=False),
                 dict(text='278.752.361', x=0.5, y=0.46, font_size=13, showarrow=False)])
fig.show()

Interpretasi
Berdasarkan https://nasional.kompas.com/read/2022/04/27/03000051/jumlah-penduduk-indonesia-2022?page=all#:~:text=Worldometer%20merilis%20data%20jumlah%20penduduk,Perserikatan%20Bangsa%2Dbangsa%20atau%20PBB.&text=Jumlah%20penduduk%20Indonesia%20merupakan%203%2C51%20persen%20dari%20total%20penduduk%20dunia. jumlah penduduk di indonesia adalah 278 juta dan baru 2.91% cakupan suntik 1 dan 2.78% cakupan suntik 2

##### Vaksinisasi WIlayah

In [264]:
df_DU_VW

No.            Kecamatan  Kota / Kabupaten  Target Total per Kecamatan  \
0     1               Cakung     Jakarta Timur                      515628   
1     2           Cengkareng     Jakarta Barat                      519118   
2     3           Kali Deres     Jakarta Barat                      406018   
3     4            Cilincing     Jakarta Utara                      392136   
4     5          Duren Sawit     Jakarta Timur                      404329   
5     6        Tanjung Priok     Jakarta Utara                      377764   
6     7            Jagakarsa   Jakarta Selatan                      329378   
7     8          Kebon Jeruk     Jakarta Barat                      327860   
8     9                 Koja     Jakarta Utara                      309976   
9    10       Kebayoran Lama   Jakarta Selatan                      299047   
10   11           Jatinegara     Jakarta Timur                      295073   
11   12          Kramat Jati     Jakarta Timur                      286172   
12   13         Pasar Minggu   Jakarta Selatan                      300472   
13   14          Penjaringan     Jakarta Utara                      292546   
14   15              Ciracas     Jakarta Timur                      286157   
15   16          Pulo Gadung     Jakarta Timur                      280197   
16   17            Kembangan     Jakarta Barat                      275438   
17   18             Cipayung     Jakarta Timur                      263441   
18   19              Tambora     Jakarta Barat                      247014   
19   20         Pesanggrahan   Jakarta Selatan                      239745   
20   21            Kemayoran     Jakarta Pusat                      231712   
21   22                Tebet   Jakarta Selatan                      219339   
22   23    Grogol Petamburan     Jakarta Barat                      218293   
23   24             Palmerah     Jakarta Barat                      209979   
24   25           Pasar Rebo     Jakarta Timur                      211912   
25   26             Cilandak   Jakarta Selatan                      206339   
26   27              Makasar     Jakarta Timur                      201749   
27   28             Matraman     Jakarta Timur                      173223   
28   29          Tanah Abang     Jakarta Pusat                      159859   
29   30             Pancoran   Jakarta Selatan                      160208   
30   31           Pademangan     Jakarta Utara                      153617   
31   32     Mampang Prapatan   Jakarta Selatan                      144281   
32   33       Kebayoran Baru   Jakarta Selatan                      143481   
33   34           Taman Sari     Jakarta Barat                      118051   
34   35          Sawah Besar     Jakarta Pusat                      113400   
35   36        Kelapa Gading     Jakarta Utara                      132180   
36   37           Johar Baru     Jakarta Pusat                      128851   
37   38                Senen     Jakarta Pusat                      116078   
38   39           Setia Budi   Jakarta Selatan                      107175   
39   40        Cempaka Putih     Jakarta Pusat                       91267   
40   41               Gambir     Jakarta Pusat                       87798   
41   42              Menteng     Jakarta Pusat                       81850   
42   43    Kep. Seribu Utara  Kepulauan Seribu                       14892   
43   44  Kep. Seribu Selatan  Kepulauan Seribu                       10673   

    Target Harian per Kecamatan  Cakupan Suntik 1      Keterangan  \
0                          5333             74966  Tidak Tercapai   
1                          5041             85668  Tidak Tercapai   
2                          4295             42991  Tidak Tercapai   
3                          4073             41494  Tidak Tercapai   
4                          4006             41413  Tidak Tercapai   
5                          3728             81521  Tidak Tercapai   
6         

In [265]:
df_DU_VW_grouped_kota

Total Vaksinasi
Kota / Kabupaten                 
Jakarta Barat             3889204
Jakarta Pusat             3905840
Jakarta Selatan           5564185
Jakarta Timur             4721845
Jakarta Utara             3348347
Kepulauan Seribu            45919

In [262]:
df_DU_VW_grouped_kota.sum()

Total Vaksinasi    21475340
dtype: int64

In [268]:
px.bar(df_DU_VW_grouped_kota,color=df_DU_VW_grouped_kota.index,text_auto=True)

### F. Daftar Pustaka

https://www.suara.com/news/2021/06/27/003810/covid-19-di-indonesia-menggila-ini-data-lonjakan-kasus-sepanjang-bulan-juni